service/internal/components/loggers.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package components // import "go.opentelemetry.io/collector/service/internal/components"
+
+import (
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+func ReceiverLogger(logger *zap.Logger, id component.ID, dt component.DataType) *zap.Logger {
+	return logger.With(
+		zap.String(ZapKindKey, ZapKindReceiver),
+		zap.String(ZapNameKey, id.String()),
+		zap.String(ZapKindPipeline, string(dt)))
+}
+
+func ProcessorLogger(logger *zap.Logger, id component.ID, pipelineID component.ID) *zap.Logger {
+	return logger.With(
+		zap.String(ZapKindKey, ZapKindProcessor),
+		zap.String(ZapNameKey, id.String()),
+		zap.String(ZapKindPipeline, pipelineID.String()))
+}
+
+func ExporterLogger(logger *zap.Logger, id component.ID, dt component.DataType) *zap.Logger {
+	return logger.With(
+		zap.String(ZapKindKey, ZapKindExporter),
```

Comment:
I think the constants can be private now.

Reply:
Done. I also pulled in the `extensionLogger` function.